packages/@lwc/engine/src/framework/decorators/register.ts

```diff
@@ -49,7 +48,7 @@ interface WireCompilerDef {
     method?: number;
     adapter: WireAdapterConstructor;
     config: ConfigCallback;
-    params?: Record<string, any>;
+    params?: string[];
```

Comment:
Bikeshedding: What do you think of dynamic instead of params because this array list only a subset of the actual wire parameters?